servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpLoadBalancerFactory.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.client.api.ConnectionFactory;
+import io.servicetalk.client.api.LoadBalancer;
+import io.servicetalk.client.api.LoadBalancerFactory;
+import io.servicetalk.client.api.ReservableRequestConcurrencyController;
+import io.servicetalk.client.api.ScoreSupplier;
+import io.servicetalk.client.api.ServiceDiscovererEvent;
+import io.servicetalk.concurrent.api.Completable;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.context.api.ContextMap;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default implementation of {@link HttpLoadBalancerFactory}.
+ *
+ * @param <ResolvedAddress> The type of address after resolution.
+ */
+public final class DefaultHttpLoadBalancerFactory<ResolvedAddress>
```

Comment:
In the past you used `DefaultHttpLoadBalancerFactory` to wire a `RequestTracker`. If we "move" this class to `http-api` and eventually remove it from `http-netty`, similar use-cases won't be possible bcz `http-api` doesn't have access to `loadbalancer` types. Doesn't mean we won't be able to create an internal alternative to `DefaultHttpLoadBalancerFactory ` in `http-netty`, just something to consider before we merge.

Reply:
I think it's fine. It's a pretty bare bones implementation and seems reasonable to have in the http-netty class. otoh, if we don't want to expand the API and introduce a deprecation we can just drop the PR and go with what's in https://github.com/apple/servicetalk/pull/2910 which is to introduce a private local copy.

Reply:
I don't mind to accept this PR. After removing builder it looks like a very thin connection layer between LB API and HTTP API. I would expect it to be useful for most cases. Only if a more feature-reach or lb-specific impl is needed in the future, we can make it part of http-netty.